Fix build

KarishmaBothara opened the pull request:
I have looked at the polkadots current UI repo at https://github.com/polkadot-js/apps to fix most of the typescript errors..
Have removed the packages we don't support yet for example staking, council, treasury.
There are two places where I ignore typescript error, one of they are staking related..which I will check while working on staking.. The other place where I have added it is for a styled component.. I am looking to fix this one.. 
But the diff is huge.. so thought to create PR while I fix that one.

KarishmaBothara commented:
Need to fix lint.. investigating the issues with yarn lint.. trying to make circle ci build happy..

alexsednz commented on packages/react-api/src/ledger.ts:
```diff
@@ -31,8 +31,14 @@ export function clearLedger (): void {
 export function getLedger (): Ledger {
   if (!ledger) {
-    ledger = new Ledger(uiSettings.ledgerConn as 'u2f');
+    const def = api && ALLOWED_CHAINS.find(([g]) => g === api.genesisHash.toHex());
```
Would allowed_chains now restrict us to kusama?

KarishmaBothara replied:
no.. I will update allowed chains to include polkadot too, as 
const ALLOWED_CHAINS: [string, 'kusama' | 'polkadot'][] = [
  [KUSAMA_GENESIS, 'kusama'],
  [POLKADOT_GENESIS, 'polkadot']
];

KarishmaBothara commented on packages/apps/src/SideBar/SideBarScroll.tsx:
```diff
@@ -7,6 +7,7 @@ const SideBarScroll = styled.div.attrs({
   @media (min-width:1000px){
     .expanded & {
       position: fixed;
+      width: 12%;
```
Updated the css property to fix the issue 
![Uploading Screen Shot 2020-10-28 at 12.41.18 PM.png…]()

KarishmaBothara commented:
Fixed all the issues mentioned [here](https://centralitydev.atlassian.net/browse/IND-360?focusedCommentId=23339) as well fixed the lint issue by adding resolution for the following packages.. 
     "@polkadot/dev": "0.40.19",
     "@babel/cli": "7.8.4",
     "@babel/code-frame": "7.8.3",
     "@babel/compat-data": "7.8.4",
     "@babel/core": "7.8.4",
     "@babel/generator": "7.8.4",

@polkadot/keyring internally installs @polkadot/dev at 0.40.22 and updated babel packages which was causing the lint issues..

alexsednz commented on packages/react-api/src/ledger.ts:
```diff
@@ -5,17 +5,32 @@
+
+const ALLOWED_CHAINS: [string, 'kusama' | 'polkadot'][] = [
+  [KUSAMA_GENESIS, 'kusama'],
+  [POLKADOT_GENESIS, 'polkadot']
```
What about Azalea and Nikau?

KarishmaBothara replied:
ya.. I am looking at https://github.com/polkadot-js/common/blob/master/packages/networks/src/index.ts package to see how they use the KUSAMA and POLKADOT networks genesis hash.. I am also trying to understand when this Ledger is been used..

KarishmaBothara replied:
Right now as we don't support hardware type...we won't add support for Azalea n Nikau at this stage.

alexsednz commented:
Looks a nice work to me as a non-expert!

holygits commented:
Working well from my tests so far.
One minor thing looks odd on the sidebar (firefox)
<img width="180" alt="Screen Shot 2020-11-03 at 13 20 25" src="https://user-images.githubusercontent.com/5133901/97933971-2de4a500-1dd9-11eb-962a-d19b89c258af.png">

AlexZhenWang commented on packages/app-explorer/src/Events.tsx:
```diff
@@ -56,7 +56,7 @@ function Events ({ emptyLabel, eventClassName, events, withoutIndex, t }: Props)
                 {
                   event.meta && event.meta.documentation
-                    ? event.meta.documentation.join(' ')
+                    ? event.meta.documentation.join(' ').split('\\[info\\]').join(' ')
```
I guess we can also use `replace(//g, '' )` to replace all, but yeah, in this case we will need to write code like `a.replace(/\\\[info\\\]/g,"rr")` to process the special character `[` and `]`, which make `a.replace(/\\\[info\\\]/g,"rr")` looks confuse T T. love your implement 👍

AlexZhenWang reviewed (approved):
LGTM~

holygits commented:
I'm in favour of merging this, small stylistic fixes can be done in future PRs
